projects/ui/src/components/Silo/Actions/Convert.tsx

```diff
@@ -364,210 +413,312 @@ const ConvertForm : FC<
+
+  const defaultFarmActionsFormState = useMemo(() => {
+    const shouldChange = true;
+    const { isLP, isUnripe } = fromToken;
+    if (!shouldChange) {
+      return {
+        preset: 'plant',
+        primary: undefined,
+        secondary: undefined,
+        implied: [FormTxn.MOW],
+      };
+    }
+
+    // only show the 'plant' option in primary options if the token is BEAN
+    const presetKey = isUnripe || isLP ? 'noPrimary' : 'plant';
+
+    return {
+      preset: presetKey,
+      primary: undefined,
+      secondary: undefined,
+    };
+  }, [fromToken]);
 
   /// Form
-  const middleware    = useFormMiddleware();
-  const initialValues : ConvertFormValues = useMemo(() => ({
-    // Settings
-    settings: {
-      slippage: 0.1,
-    },
-    // Token Inputs
-    tokens: [
-      {
-        token:      fromToken,
-        amount:     undefined,
-        quoting:    false,
-        amountOut:  undefined,
+  const middleware = useFormMiddleware();
+  const initialValues: ConvertFormValues = useMemo(
+    () => ({
+      // Settings
+      settings: {
+        slippage: 0.1,
       },
-    ],
-    // Convert data
-    maxAmountIn:    undefined,
-    // Token Outputs
-    tokenOut:       initialTokenOut,
-  }), [fromToken, initialTokenOut]);
+      // Token Inputs
+      tokens: [
+        {
+          token: fromToken,
+          amount: undefined,
+          quoting: false,
+          amountOut: undefined,
+        },
+      ],
+      // Convert data
+      maxAmountIn: undefined,
+      // Token Outputs
+      tokenOut: initialTokenOut,
+      farmActions: defaultFarmActionsFormState,
+    }),
+    [defaultFarmActionsFormState, fromToken, initialTokenOut]
+  );
+
+  /**
+   * SDK.SILO.CONVERT.convertEstimate has all the logic for this,
+   * but we need to add additional crates if the user is planting
+   */
+  const handleConversion = useCallback(
+    async (
+      tokenIn: Token,
+      _amountIn: BigNumber,
+      tokenOut: Token,
+      slippage: number,
+      isPlanting: boolean
+    ) => {
+      if (!account) {
+        throw new Error('Signer required');
+      }
+
+      const siloConvert = sdk.silo.siloConvert;
+      // we define this here b/c siloConvert expects token instances from it's own class
+      const whitelist = [
+        siloConvert.Bean,
+        siloConvert.BeanCrv3,
+        siloConvert.urBean,
+        siloConvert.urBeanCrv3,
+      ];
+
+      const [inToken, outToken] = whitelist.reduce(
+        (prev, curr) => {
+          prev[0] = curr.equals(tokenIn) ? curr : prev[0];
+          prev[1] = curr.equals(tokenOut) ? curr : prev[1];
+          return prev;
+        },
+        [null, null] as [Token | null, Token | null]
+      );
+
+      if (!inToken || !outToken) throw new Error('conversion unavailable');
+
+      const amountIn = inToken.amount(_amountIn.toString());
+      const { deposited } = await sdk.silo.getBalance(inToken, account, {
+        source: DataSource.LEDGER,
```

Comment:
This is true, but the only reason why I couldn't use sdk.silo.siloConvert.convertEstimate is because the function checks the balance of the user's silo deposits and will fail if I pass in an additional crate for what will be converted via planting before